### ISSUE TITLE
[Issue] Windows OCI images are part of rancher-images.txt / assets

### ISSUE THREAD

**Issue by smangelkramer** [rancher/rancher#39510]
**Rancher Server Setup**
- Rancher version: v2.6.9 (also in older releases)
- Air Gapped / Private Registry / Image pulling issue

**Describe the issue**

Two (Windows) OCI images are part of the assets  of the current (stable) Rancher release (v2.6.9):

- `rancher-images.txt` ( https://github.com/rancher/rancher/releases/download/v2.6.9/rancher-images.txt )
- `rancher-images-digests-linux-amd64.txt` (https://github.com/rancher/rancher/releases/download/v2.6.9/rancher-images-digests-linux-amd64.txt )
Images:

```
docker.io/rancher/pause:3.6-windows-1809-amd64 
docker.io/rancher/pause:3.6-windows-ltsc2022-amd64 
```

When pulling the images for this release for an AirGapped setup with private registry,
the following error occurs during pulling the images (on a Linux x86 machine):

```
>> Pulling image: rancher/pause:3.6-windows-1809-amd64
3.6-windows-1809-amd64: Pulling from rancher/pause
bc8517709e9c: Pulling fs layer
9b4754cb2394: Pulling fs layer
6851284b6a3b: Pulling fs layer
f3395efd3035: Pulling fs layer
f3395efd3035: Waiting
6851284b6a3b: Verifying Checksum
6851284b6a3b: Download complete
9b4754cb2394: Verifying Checksum
9b4754cb2394: Download complete
image operating system "windows" cannot be used on this platform
Error: >>> Image pull failed: rancher/pause:3.6-windows-1809-amd64
Command exited with non-zero status 1
```

As a workaround, I removed the two OCI images from the list and was able to cleanly pull all the remaining images. 
The question now is whether this is intentional or an oversight?

When using the rancher-load/save scripts from the release, this is not noticeable at first. 
I noticed it when I automated the pull and push in the local registry via a CI job.

Here is an example with the official scripts:

```
ubuntu@sm-jumphost:~/rancher-images$ ./rancher-save-images.sh
...
Image pull success: rancher/pause:3.6
Image pull failed: rancher/pause:3.6-windows-1809-amd64
Image pull failed: rancher/pause:3.6-windows-ltsc2022-amd64
Image pull success: rancher/pipeline-jenkins-server:v0.1.4
Creating rancher-images.tar.gz with 2 images
...
```

Sure, you can get around this in a pipeline, but it's not nice and clean from my point of view.
I use `set -euo pipefail` in my pipelining scripts, for me this "clean" because certain or 
common issues will cause the script to explicitly fail (loudly) - and this prevents "hidden" issues in production later.

**To Reproduce**

Do a simple `docker pull` on any linux machine for one of the two images named above.

**Result**

**Expected Result**

You will run into this error message:

```
ubuntu@sm-jumphost:~$ docker pull rancher/pause:3.6-windows-ltsc2022-amd64
3.6-windows-ltsc2022-amd64: Pulling from rancher/pause
ee7a9c9193e3: Pulling fs layer
784aba008ad8: Download complete
02e02a05e018: Pulling fs layer
0d37d6f9240b: Waiting
image operating system "windows" cannot be used on this platform
```

SURE-4711

**Comment by markusewalker** [comment]
**QA TEST PLAN**

**Scenarios**
|Scenario|Test Case|
|---|---|
|1|In an air-gapped setup with a private registry, pull the Windows OCI images defined in the issue for 2.6.9|
|2|In an air-gapped setup with a private registry, run the `rancher-save-images.sh` script|

**Comment by markusewalker** [comment]
Reproduced this using the `v2.7.0` Windows OCI images. See details below:

**ENVIRONMENT DETAILS**
- Rancher install: Air-gapped Ubuntu 20.04 AMI with private registry
- Rancher images version:
     - Reproduced: `v2.6.9`

**TEST RESULT**
|Scenario|Test Case|Result
|---|---|---|
|1|In an air-gapped setup with a private registry, pull the Windows OCI images defined in the issue for 2.7.0|:x:|
|2| N/A as test scenario one failed, so this will also fail|:x:|

**REPRODUCTION STEPS**

**Scenario 1**
1. Navigated to the air-gapped environment that has a private registry in the terminal.
2. Ran command `docker pull rancher/pause:3.6-windows-1809-amd64` and received the following result:
```
$ docker pull rancher/pause:3.6-windows-1809-amd64 
3.6-windows-1809-amd64: Pulling from rancher/pause
bc8517709e9c: Pulling fs layer 
9b4754cb2394: Download complete 
6851284b6a3b: Pulling fs layer 
f3395efd3035: Waiting 
image operating system "windows" cannot be used on this platform
```
3. Ran command `docker pull rancher/pause:3.6-windows-ltsc2022-amd64` and received the following result:
```
$ docker pull rancher/pause:3.6-windows-ltsc2022-amd64 
3.6-windows-ltsc2022-amd64: Pulling from rancher/pause
ee7a9c9193e3: Pulling fs layer 
784aba008ad8: Download complete 
02e02a05e018: Downloading [>                                                  ]  14.39kB/1.402MB
0d37d6f9240b: Waiting 
image operating system "windows" cannot be used on this platform
```

**Comment by jiaqiluo** [comment]
Can be validated in Rancher [v2.7.2-rc1](https://github.com/rancher/rancher/releases/tag/v2.7.2-rc1)

**Comment by markusewalker** [comment]
Verified this is resolved on `v2.7.2-rc1`. Details below:

**ENVIRONMENT DETAILS**
- Rancher install: Air-gapped Ubuntu 20.04 AMI with private registry
- Rancher images version:
     - Reproduced: `v2.7.2-rc1`

**TEST RESULT**
|Scenario|Test Case|Result
|---|---|---|
|1|In an air-gapped setup with a private registry, pull the Windows OCI images defined in the issue for 2.7.2-rc1|:white_check_mark:|
|2| N/A as test scenario one failed, so this will also fail|:white_check_mark:|

**VALIDATION STEPS**

**Scenario 1**
1. Navigated to the air-gapped environment that has a private registry in the terminal.
2. Ran command `docker pull rancher/pause:3.6-windows-1809-amd64` and received the following expected result:
```
$ docker pull rancher/pause:3.6-windows-1809-amd64 
3.6-windows-1809-amd64: Pulling from rancher/pause
bc8517709e9c: Pulling fs layer 
9b4754cb2394: Download complete 
6851284b6a3b: Pulling fs layer 
f3395efd3035: Waiting 
image operating system "windows" cannot be used on this platform
```
3. Ran command `docker pull rancher/pause:3.6-windows-ltsc2022-amd64` and received the following expected result:
```
$ docker pull rancher/pause:3.6-windows-ltsc2022-amd64 
3.6-windows-ltsc2022-amd64: Pulling from rancher/pause
ee7a9c9193e3: Pulling fs layer 
784aba008ad8: Download complete 
02e02a05e018: Downloading [>                                                  ]  14.39kB/1.402MB
0d37d6f9240b: Waiting 
image operating system "windows" cannot be used on this platform
```
4. Ran command `grep -i windows rancher-save-images.txt` and validated that the only results is the following:
```
$ grep -i "windows" rancher-images.txt 
rancher/mirrored-sigwindowstools-k8s-gmsa-webhook:v0.3.0
```
________
**Scenario 2**
1. Navigated to the air-gapped environment that has a private registry in the terminal.
2. Downloaded the `rancher-save-images.sh` from the `v2.7.2-rc1` release assets.
3. Ran `chmod +x rancher-save-images.sh` and ran `./rancher-save-images.sh`.
4. I was successfully able to download all images in the `rancher-save-images.txt` without failure.
